### PR TITLE
Fixes for 58176: Middle clicking files in breadcrumbs explorer opens …

### PR DESCRIPTION
**Pull request by J-Graham**
…as preview

Added a variable to detect the middle click and pass the pinned option when this is true

### CODE REVIEW

**Comment by J-Graham** [comment]
This is a PR for the following issue: https://github.com/Microsoft/vscode/issues/58176.  It's my first attempt and I would love to hear any feedback on if you felt there was a better way to implement this.

File: src/vs/workbench/browser/parts/editor/breadcrumbsControl.ts [hunk 1]
```diff
@@ -325,7 +325,7 @@ export class BreadcrumbsControl {
 						editorViewState = undefined;
 					}
 					this._contextViewService.hideContextView(this);
-					this._revealInEditor(event, data.target, this._getEditorGroup(data.payload && data.payload.originalEvent));
+					this._revealInEditor(event, data.target, this._getEditorGroup(data.payload && data.payload.originalEvent), data.payload.originalEvent.middleButton);
```
Comment (jrieken) [inline src/vs/workbench/browser/parts/editor/breadcrumbsControl.ts]: `data.payload` might be undefined and will explode now

File: src/vs/workbench/browser/parts/editor/breadcrumbsControl.ts [hunk 2]
```diff
@@ -408,11 +408,11 @@ export class BreadcrumbsControl {
 		this._ckBreadcrumbsActive.set(value);
 	}
 
-	private _revealInEditor(event: IBreadcrumbsItemEvent, element: any, group: SIDE_GROUP_TYPE | ACTIVE_GROUP_TYPE): void {
+	private _revealInEditor(event: IBreadcrumbsItemEvent, element: any, group: SIDE_GROUP_TYPE | ACTIVE_GROUP_TYPE, middleClick: boolean = false): void {
```
Comment (jrieken) [inline src/vs/workbench/browser/parts/editor/breadcrumbsControl.ts]: I'd call it `pinned` instead of `middleClick`

**Review by jrieken — CHANGES_REQUESTED** [review]

**Comment by J-Graham** [comment]
Thanks for the feedback @jrieken.  Let me know if these changes cover what you need.

**Review by jrieken — APPROVED** [review]
lgtm

**Comment by jrieken** [comment]
Thanks

**Pull request merged**